### PR TITLE
Changes made to allow deployment to Heroku

### PR DESCRIPTION
**Pull request by sibsmc**
This PR adds settings and gems needed to run the application on Heroku.

## Background

It was found that the frontend teams were interested in having a deployed and try-outable version of the server platform.

Heroku deployment is easy, so we went for  that.

## Solution

- Add necessary gems
- Update configuration files

### CODE REVIEW

File: server/Gemfile [hunk 1]
```diff
@@ -3,15 +3,25 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.6.5'
 
+#gem 'dotenv-rails', groups: [:development, :test]
```
Comment (olleolleolle) [inline server/Gemfile]:
```suggestion
```

Commented-out code, can be removed.

File: server/Gemfile [hunk 2]
```diff
@@ -33,9 +43,13 @@ group :development, :test do
 
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
-  gem 'rspec-rails', '~> 3.9'
+  # Adds support for Capybara system testing and selenium driver
+  gem 'pry-rescue'
+  gem 'pry-stack_explorer'
+  gem 'capybara', '~> 3.31'
+  gem 'selenium-webdriver'
```
Comment (olleolleolle) [inline server/Gemfile]: All of these gems  can be added as we go along, right? By adding them, we prolong the install cycle in Travis CI by quite a bit, perhaps we can add them in when we add the first Capybara test?

**Comment by sibsmc** [comment]
Started as a small number of changes to deploy app to heroku, ended up needing to update ruby version so a lot of files have changed.

File: server/package.json [hunk 3]
```diff
@@ -2,4 +2,7 @@
   "name": "mentor-mentee-platform",
   "private": true,
   "dependencies": {}
+  "engines": {
```
Comment (olleolleolle) [inline server/package.json]: Line 4 is missing a comma at end-of-line, which makes this JSON invalid.

**Pull request merged**